From: Loic Minier <lool@dooz.org>
Date: Sun, 21 Oct 2007 22:05:42 +0200
Subject: Don't list images from unknown directories in icon cache

After GTK 2.12.0, gtk-update-icon-cache fails if there is a PNG file
placed directly in /usr/share/icons/hicolor (not in a correct
subdirectory like 48x48/apps).

TODO: This is believed to have been fixed differently upstream, so maybe
this change is no longer necessary:
"I believe a slightly different fix that I did some time ago fixes this
too." --Matthias Clasen, 2008-02-16 04:43:10 UTC

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=451634
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=444285
Forwarded: yes
Applied-upstream: no
---
 gtk/updateiconcache.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/updateiconcache.c b/gtk/updateiconcache.c
index 39e1345..9fecf12 100644
--- a/gtk/updateiconcache.c
+++ b/gtk/updateiconcache.c
@@ -679,7 +679,7 @@ scan_directory (const gchar *base_path,
 		      directories = g_list_append (directories, g_strdup (subdir));
 		    }
 		  else
-		    dir_index = 0xffff;
+		    continue;
 		}
 		
 	      image = g_new0 (Image, 1);
